include/mbedtls/ssl.h

```diff
@@ -2220,6 +2226,27 @@ void mbedtls_ssl_set_timer_cb( mbedtls_ssl_context *ssl,
                                mbedtls_ssl_set_timer_t *f_set_timer,
                                mbedtls_ssl_get_timer_t *f_get_timer );
 
+#if defined(MBEDTLS_SSL_SRV_C)
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION)
+/**
+ * \brief           Set the certificate selection callback (server-side only).
+ *
+ *                  If set, the callback is always called for each handshake,
+ *                  after `ClientHello` processing has finished.
+ *
+ *                  The callback has the following parameters:
+ *                  - \c mbedtls_ssl_context*: The SSL context to which
+ *                                             the operation applies.
+ *
```

Comment:
Please describe what the return value form the callback means.

Reply:
:rocket:

Reply:
you can do this with \return or \retval after the \param (s) as well, as you did with the other functions.

Reply:
I see `\return` or `\returns` in mbedtls/ssl.h is used to document the return value of the functions each is describing, not the callback fn when a callback fn is an argument to the function being documented.  The only places in mbedtls/ssl.h where `\return` is used to describe the callback fn return value is in typedefs for such callback functions (near the top of mbedtls/ssl.h).  Would you please point me at doc for the markup?  I have been doing my best to mimic the existing formats use that I see.

Reply:
I think the way you did it is correct and matching existing usage. As a matter of personal preference, I like to have a separate typedef for the callback type, but we don't have any agreed upon guidelines about that, and existing usage is split, so you can do it either way.

Also, as a user of Mbed TLS, you're more used than us to reading our documentation with fresh eyes, so if you have an opinion as to which presentation is better, feel free to weigh in.

Reply:
I agree that's beyond the scope of this PR.